src/store_tests.cc

```diff
@@ -98,17 +99,21 @@ bool test_protect(bool print_all) {
     print_bytes(size_unencrypted_data, unencrypted_data);
     printf("\n");
   }
+  
+  // Unprotect should return the key used for protecting the data
   if (!same_key(key_start, key_end))
     return false;
+
+  // Unencrypted data should come out same as what was fed-in to protect
   if (memcmp(unencrypted_data, (byte*)secret_data, strlen(secret_data)) != 0)
     return false;
 
-  key_message new_key;
+  key_message new_key; // ? RESOLVE: This is uninit'ed. Don't we want to supply a new key here?
```

Comment:
Let's discuss this one ... I think the interface is wonky ... and not sure why 'key' is even needed.